### PR TITLE
common, prov/tcp: Report more detail info about chosen addr

### PR DESCRIPTION
**Pull request by nikitaxgusev**

### CODE REVIEW

**Comment by nikitaxgusev** [comment]
@shefty If the approch of adding `ofi_chosen_ipaddr()` looks good to you, we can add the call to other providers.

File: src/common.c [hunk 1]
```diff
@@ -1154,6 +1154,28 @@ ofi_addr_list_entry_comp_speed(struct slist_entry *cur, const void *insert)
 	return (cur_addr->speed < insert_addr->speed);
 }
 
+void ofi_chosen_ipaddr(const struct fi_provider *prov,
+			 struct ofi_addr_list_entry *addr_entry)
```
Comment (shefty) [inline src/common.c]: addr_list_entry contains an ipstr member.  Can the caller just call FI_INFO directly to print that value?

Comment (nikitaxgusev) [reply src/common.c]: @shefty Yes, it can. Changed.
Does the sockets provider require this approach?

Comment (shefty) [reply src/common.c]: We're trying not to change the sockets provider at this point, unless it's critical.  We want to move completely to tcp.

Comment (nikitaxgusev) [reply src/common.c]: @shefty Ok. Would you like to see the ofi_chosen_ipaddr on other ones?

Comment (shefty) [reply src/common.c]: No.  I don't see that we need this function at all. It's now a 1-line print statement.

Comment (nikitaxgusev) [reply src/common.c]: @shefty Done.

**Pull request merged**